tests/test_path.rs

```diff
@@ -50,3 +50,17 @@ fn parse_interpolated_leading_component() {
     }
     "###);
 }
+
+#[test]
+fn print_incomplete_qpath() {
+    use quote::ToTokens;
+    use syn::Type;
+
+    let mut ty: Type = syn::parse_quote!(<Self as A>::Q);
+    if let Type::Path(tp) = &mut ty {
```

Comment:
Recommend parsing directly to TypePath to bypass the unsuccessful branch of this `if let`.

```suggestion
    let mut ty: TypePath = syn::parse_quote!(<Self as A>::Q);
```